src/Application/ApplicationFileProcessor.php

```diff
@@ -161,11 +182,43 @@ public function processFiles(
         return new ProcessResult($systemErrors, $fileDiffs);
     }
 
-    private function processFile(File $file, Configuration $configuration): FileProcessResult
+    /**
+     * @param string[] $possibleDependentFiles
+     * @param string[] $filePaths
+     * @return string[]
+     */
+    private function findDependentFiles(array $possibleDependentFiles, array $filePaths): array
+    {
+        $filesToAnalise = array_flip($filePaths);
+
+        $dependentFiles = [];
+        foreach ($possibleDependentFiles as $possibleDependentFile) {
+            if (! isset($filesToAnalise[$possibleDependentFile])) {
+                $fileDependencies = $this->fileDependenciesCache->getFileDependencies($possibleDependentFile);
+
+                if ($fileDependencies !== null) {
+                    foreach ($fileDependencies as $fileDependency) {
+                        if (isset($filesToAnalise[$fileDependency])) {
+                            $dependentFiles[] = $possibleDependentFile;
+                            $this->changedFilesDetector->invalidateFile($possibleDependentFile);
```

Comment:
And we also remove it from the cache, to make sure that it is analysed in future runs